src/Compatibility/ApiDiff/Microsoft.DotNet.ApiDiff.Tool/Program.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.Diagnostics;
+using Microsoft.DotNet.ApiSymbolExtensions.Logging;
+
+namespace Microsoft.DotNet.ApiDiff.Tool;
+
+/// <summary>
+/// Entrypoint for the genapidiff tool, which generates a markdown diff of two
+/// different versions of the same assembly, using the specified command line options.
+/// </summary>
+public static class Program
+{
+    public static async Task Main(string[] args)
+    {
+        RootCommand rootCommand = new("genapidiff");
+
+        Option<string> optionBeforeAssembliesFolderPath = new(["--before", "-b"])
```

Comment:
General thought - do the inputs match our other compatibility tools?  Do they accept the same sort of things in the same format as other tools?

Suppose I might want to automate a diff in a library project I build.  Would I have to do unnatural things like copy my references to a single folder?

Reply:
GenAPI/ApiCompat take as argument an array of folders. This tool only takes one folder.

The AssemblySymbolLoader `AddReferenceSearchPaths` method adds this single folder, and later the `LoadFromPaths` method enumerates the files, but it is not using `SearchOption.AllDirectories` so it won't recurse.

I see what you mean, I think we can easily change the behavior of the tool to take multiple folders, or visit recursively, if we find the need to do so. At the moment I haven't had the need to do that because the ref nupkgs contain all the DLLs in one single folder.

Reply:
I was imagining someone might want to introduce this tool into their project build.  For example WinForms, EF, Aspire, etc.  Imagine that after running APICompat they also get a nice diff for release notes.  IMO it's easier for us to be consistent.  Being arbitrarily different creates friction.  Consider as a follow up.

Reply:
Yeah you're right. I fixed it. It matches the other tools now.